src/cljs/commiteth/common.cljs

```diff
@@ -12,16 +12,24 @@
                    :on-change #(reset! val-ratom (-> % .-target .-value))})]))
 
 (defn dropdown [props title val-ratom items]
-  (fn []
-    (if (= 1 (count items))
-      (reset! val-ratom (first items)))
-    [:select.ui.basic.selection.dropdown
-     (merge props {:on-change
-                   #(reset! val-ratom (-> % .-target .-value))})
-     (doall (for [item items]
-              ^{:key item} [:option
-                            {:value item}
-                            item]))]))
+  "If val-ratom is set, preselect it in the dropdown.
+   Add value of val-ratom if it's missing from items list.
+   Otherwise, prepend title as a disabled option"
+  (let [items (cond-> items
+                  (and @val-ratom
+                       (not (contains? (set items) @val-ratom)))
+                  (conj @val-ratom)
+                  (not @val-ratom)
+                  (into [title]))]
```

Comment:
Actually `into` has the same semantics as `conj`, since I think that it uses `conj` in the end.

By the way, instead of dealing with the semantics of clj collections, how about just generating the `option` html element before the `for` for the title when it is needed?

Reply:
`into` will invoke `conj` on a vector `[title]`, so it will nicely put all seq items at the end, so no problem here.

Reply:
Yeah, that treatment of collections is quite tricky in Clojure and is probably one of the things that it has to improve. :S

Btw, when possible could you please send a new comment instead of edit/update the previous one? To ensure that notifications are also sent for updates. Thanks! :)

Reply:
Got you, ok! I thought notifications are delivered for comment updates as well.